detection_rules/rule.py

```diff
@@ -700,6 +700,13 @@ class QueryRuleData(BaseRuleData):
     language: definitions.FilterLanguages
     alert_suppression: Optional[AlertSuppressionMapping] = field(metadata=dict(metadata=dict(min_compat="8.8")))
 
+    @cached_property
+    def index_or_dataview(self) -> list[str]:
```

Comment:
`index_or_dataview` is a bit confusing when combined with the return value of a list. Is it a list of indices or data views? Should it be a singular value instead of a list?

Reply:
if changing it is a big thing, we can leave it as is and come back to it later

Reply:
I agree, generally speaking I would prefer it not to be a list; however, the purpose of having it be a list is to avoid having to type check (as the iteration check would detect a `None` value) [context](https://github.com/elastic/detection-rules/pull/3679#pullrequestreview-2064609443).